feature/experimental/system/gnmi/benchmarking/ate_tests/drained_configuration_convergence_time/drained_configuration_convergence_time_bgp_test.go

```diff
@@ -0,0 +1,264 @@
+func verifyBGPAsPath(t *testing.T, dut *ondatra.DUTDevice) {
+	ate := ondatra.ATE(t, "ate")
+
+	dutPolicyConfPath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).
+		Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp().
+		PeerGroup(setup.PeerGrpName).ApplyPolicy().ExportPolicy()
+
+	// Build wantAsPath to compare the diff
+	var wantAsPath []uint32
+	for i := 0; i < setup.RouteCount; i++ {
+		wantAsPath = append(wantAsPath, setup.DUTAs, setup.DUTAs, setup.DUTAs, setup.DUTAs, setup.ATEAs2)
+	}
+
+	// Start the timer.
+	start := time.Now()
+	gnmi.Replace(t, dut, dutPolicyConfPath.Config(), []string{setAspathPrependPolicy})
+	t.Run("BGP-AS-PATH Verification", func(t *testing.T) {
+		at := gnmi.OC()
+		for _, ap := range ate.Ports() {
+			if ap.ID() == "port1" {
+				//port1 is ingress, skip verification on ingress port
+				continue
+			}
+
+			statePath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).
+				Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+		prefixLoop:
+			for repeat := 4; repeat > 0; repeat-- {
+				prefixesv4 := statePath.Neighbor(setup.ATEIPList[ap.ID()].String()).
+					AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Prefixes()
+				gotSent := gnmi.Get(t, dut, prefixesv4.Sent().State())
+				switch {
+				case gotSent == setup.RouteCount:
+					t.Logf("Prefixes sent from ingress port are learnt at ATE dst port : %v are %v", setup.ATEIPList[ap.ID()].String(), setup.RouteCount)
+					break prefixLoop
+				case repeat > 0 && gotSent < setup.RouteCount:
+					t.Logf("All the prefixes are not learnt , wait for 5 secs before retry.. got %v, want %v", gotSent, setup.RouteCount)
+					time.Sleep(time.Second * 5)
+				case repeat == 0 && gotSent < setup.RouteCount:
+					t.Errorf("sent prefixes from DUT to neighbor %v is mismatch: got %v, want %v", setup.ATEIPList[ap.ID()].String(), gotSent, setup.RouteCount)
+				}
+			}
+
+			rib := at.NetworkInstance(ap.Name()).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "0").Bgp().Rib()
+			prefixPath := rib.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Ipv4Unicast().
+				NeighborAny().AdjRibInPre().RouteAny().WithPathId(0).Prefix()
+			pref := gnmi.GetAll(t, ate, prefixPath.State())
+			asPath := gnmi.GetAll(t, ate, rib.AttrSetAny().AsSegmentAny().State())
+
+			var gotAsPath []uint32
+			for _, v := range asPath {
+				gotAsPath = append(gotAsPath, v.GetMember()...)
+			}
+			if diff := cmp.Diff(wantAsPath, gotAsPath); diff != "" {
+				t.Errorf("obtained AS path on ATE is not as expected, got %v, want %v, prefixes %v", gotAsPath, wantAsPath, pref)
+			}
+		}
+	})
+
+	// End the timer and calculate time
+	elapsed := time.Since(start)
+	t.Logf("Duration taken to apply as path prepend policy is  %v", elapsed)
+}
+
+// verifyBGPSetMED is to Validate MED attribute using bgp rib telemetry on ATE
+func verifyBGPSetMED(t *testing.T, dut *ondatra.DUTDevice) {
+	ate := ondatra.ATE(t, "ate")
```

Comment:
accept ate as argument rather than looking it up again

Reply:
I have updated with similar changes for all functions. Please check latest code. Thank you.